api/jobset/v1alpha2/jobset_webhook.go

```diff
@@ -52,11 +52,15 @@ func (js *JobSet) Default() {
 			js.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode = completionModePtr(batchv1.IndexedCompletion)
 		}
 		// Enable DNS hostnames by default.
-		if js.Spec.ReplicatedJobs[i].Network == nil {
-			js.Spec.ReplicatedJobs[i].Network = &Network{}
+		if js.Spec.Network == nil {
+			js.Spec.Network = &Network{}
 		}
-		if js.Spec.ReplicatedJobs[i].Network.EnableDNSHostnames == nil {
-			js.Spec.ReplicatedJobs[i].Network.EnableDNSHostnames = pointer.Bool(true)
+		if js.Spec.Network.EnableDNSHostnames == nil {
+			js.Spec.Network.EnableDNSHostnames = pointer.Bool(true)
+		}
+		// Subdomain defaults to the JobSet name
+		if js.Spec.Network.Subdomain == "" {
+			js.Spec.Network.Subdomain = js.Name
```

Comment:
I can't comment below, but in ValidateCreate, we need to validate that the value is a valid domain name:

```
import 	"k8s.io/apimachinery/pkg/util/validation"
if js.Spec.Network != nil and js.Spec.Network.Subdomain != "" {
  allErrs = append(allErrs, validation.IsDNS1123Subdomain(subdomain));
}
```

Reply:
Changes coming soon! I'm running all the tests once more as a sanity check everything is still ok, and umm, I might have taken a dance break. :laughing: :man_dancing: